### DISCUSSION TITLE
Please, help me!

### DISCUSSION THREAD

**Discussion by fabriciohausman** [MagicFoundation/Alcinoe#220] [Q&A]
I created an app in Delphi using Alcinoe components. The app is already working and has already been published on Google Play. Now when I went to compile for iOS I can't and I get the following message: [DCC Fatal Error] iOSapi.Foundation.pas(6593): F2051 Unit System.IOUtils was compiled with a different version of iOSapi.Foundation.NSFileManager.
I've tried everything and it didn't work. Can you help me?

**Comment by fabriciohausman** [comment]
I managed to get around the previous problem. I downloaded and installed the library again and now when I try to compile for iOS I get this message: [DCC Error] ALFmxGraphics.pas(3520): E2003 Undeclared identifier: 'setValueforKey'

**Comment by Zeus64** [comment]
Hi Fabricio, please be sure to include in your search path https://github.com/Zeus64/alcinoe/tree/master/embarcadero/sydney/10_4_2 (and don't forget to call update.bat)

**Comment by fabriciohausman** [comment]
Thanks for answering.
First of all, thank you for building this library!

I applied the patch. I built an entire project using Alcinoe and it works perfectly on android but when I went to compile for iOS it happened.
I will list all errors that occur:
1. ALFmxFilterEffects TContextShaderArch.Metal > FMX.Types3D TContextShaderArch = (Undefined, DX9, DX10, DX11_level_9, DX11, GLSL, Mac, IOS, Android) here there is no "metal";
2. ALFmxGraphics -> LClampFilter.setValueforKey
3. ALFmxEdit -> FTextField.View.setAttributedPlaceholder(LTextPromptAttr);

I commented out all these lines and so it compiles.

Can you help me solve it?

**Comment by fabriciohausman** [comment]
I'm using Delphi 10.3.3
